search/search_integration_tests/smoke_test.cpp

```diff
@@ -215,64 +218,124 @@ UNIT_CLASS_TEST(SmokeTest, CategoriesTest)
       {"highway", "secondary_link"},
       {"highway", "service"},
       {"highway", "steps"},
-      {"area:highway", "steps"},
       {"highway", "tertiary"},
       {"highway", "tertiary_link"},
       {"highway", "track"},
       {"highway", "traffic_signals"},
       {"highway", "trunk"},
       {"highway", "trunk_link"},
       {"highway", "unclassified"},
+      {"historic", "citywalls"},
+  };
+  set<uint32_t> notPointTypes;
+  for (auto const & tags : arrNotPoint)
+    notPointTypes.insert(cl.GetTypeByPath(tags));
+
+  // No point drawing rules for country scale range.
+  base::StringIL const arrInvisible[] = {
       {"man_made", "tower"},
       {"man_made", "water_tower"},
-      {"man_made", "water_well"},
-      {"natural", "glacier"},
-      {"natural", "water", "pond"},
       {"natural", "tree"},
 
-      /// @todo Not a point feature. Should rewrite test.
-      {"historic", "citywalls"},
+      {"place", "continent"},
+      {"place", "county"},
+      {"place", "region"},
   };
   set<uint32_t> invisibleTypes;
-  for (auto const & tags : invisibleAsPointTags)
+  for (auto const & tags : arrInvisible)
     invisibleTypes.insert(cl.GetTypeByPath(tags));
 
-  base::StringIL const notSupportedTags[] = {
-      // Not visible for country scale range.
-      {"place", "continent"},
-      {"place", "region"}
+  // Not indexed types for Features with empty names.
+  base::StringIL const arrNoEmptyNames[] = {
+      {"area:highway"},
+      {"building"},
+      {"highway", "motorway_junction"},
+      {"landuse"},
+      {"man_made", "chimney"},
+      {"man_made", "tower"},
+      {"natural"},
+      {"office"},
+      {"place"},
+      {"waterway"},
+
+      /// @todo Controversial here.
+      /// Don't have point drawing rules except text -> type will be removed for Feature with empty name.
+      {"amenity", "brothel"},
+      {"amenity", "public_bookcase"},
+      {"building", "train_station"},
+      {"leisure", "track"},
+      {"natural", "beach"},
```

Comment:
If I search for a beach category, will I find beaches without a name?

Reply:
You will for area beaches.
This is about point styles, we don't create Features that you can't even see on map.

Reply:
Point beach should have a beach icon and be searchable too, no?